packages/workbox-core/src/types.ts

```diff
@@ -48,12 +48,23 @@ export interface RouteMatchCallback {
  * Options passed to a `RouteHandlerCallback` function.
  */
 export interface RouteHandlerCallbackOptions {
-  request: Request | string;
-  url?: URL;
+  event: ExtendableEvent;
+  request: Request;
+  url: URL;
   params?: string[] | MapLikeObject;
+}
+
+/**
+ * Options passed to a `ManualHandlerCallback` function.
+ */
+export interface ManualHandlerCallbackOptions {
```

Comment:
Yup.

Reply:
Actually, no, because `ManualHandlerCallbackOptions` defines `request: Request | string;` and we want `RouteHandlerCallbackOptions` to be `request: Request;`. There are enough different about the two that it would be easier not to extend and override.

Reply:
Gotcha.